Content.Client/Radiation/Overlays/RadiationDebugOverlay.cs

```diff
@@ -0,0 +1,167 @@
+
+namespace Content.Client.Radiation.Overlays;
+
+public sealed class RadiationDebugOverlay : Overlay
+{
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    private readonly Font _font;
+
+    public List<RadiationRay>? Rays;
+    public Dictionary<EntityUid, Dictionary<Vector2i, float>>? ResistanceGrids;
+
+    public override OverlaySpace Space => OverlaySpace.WorldSpace | OverlaySpace.ScreenSpace;
+
+    public RadiationDebugOverlay()
+    {
+        IoCManager.InjectDependencies(this);
+
+        var cache = IoCManager.Resolve<IResourceCache>();
+        _font = new VectorFont(cache.GetResource<FontResource>("/Fonts/NotoSans/NotoSans-Regular.ttf"), 8);
+    }
+
+    protected override void Draw(in OverlayDrawArgs args)
+    {
+        switch (args.Space)
+        {
+            case OverlaySpace.ScreenSpace:
+                DrawScreenRays(args);
+                DrawScreenResistance(args);
+                break;
+            case OverlaySpace.WorldSpace:
+                DrawWorld(args);
+                break;
+        }
+    }
+
+    private void DrawScreenRays(OverlayDrawArgs args)
+    {
+        if (Rays == null || args.ViewportControl == null)
+            return;
+
+        var handle = args.ScreenHandle;
+        foreach (var ray in Rays)
+        {
+            if (ray.MapId != args.MapId)
+                continue;
+
+            if (ray.ReachedDestination)
+            {
+                var screenCenter = args.ViewportControl.WorldToScreen(ray.Destination);
+                handle.DrawString(_font, screenCenter, ray.Rads.ToString("F2"), 2f, Color.White);
+            }
+
+            foreach (var (blockerPos, rads) in ray.Blockers)
+            {
+                var screenCenter = args.ViewportControl.WorldToScreen(blockerPos);
+                handle.DrawString(_font, screenCenter, rads.ToString("F2"), 1.5f, Color.White);
+            }
+
+            if (_mapManager.TryGetGrid(ray.Grid, out var grid))
+            {
+                foreach (var (tile, rads) in ray.VisitedTiles)
+                {
+                    if (rads == null)
+                        continue;
+
+                    var worldPos = grid.GridTileToWorldPos(tile);
+                    var screenCenter = args.ViewportControl.WorldToScreen(worldPos);
+                    handle.DrawString(_font, screenCenter, rads.Value.ToString("F2"), 1.5f, Color.White);
+                }
+            }
+        }
+    }
+
+    private void DrawScreenResistance(OverlayDrawArgs args)
+    {
+        if (ResistanceGrids == null || args.ViewportControl == null)
+            return;
+
+        var handle = args.ScreenHandle;
+        foreach (var (gridUid, resMap) in ResistanceGrids)
+        {
+            if (!_mapManager.TryGetGrid(gridUid, out var grid))
+                continue;
+            if (grid.ParentMapId != args.MapId)
+                continue;
```

Comment:
Use transformcomp's mapid (I say this because I want to nuke ParentMapId because it does a lot of silly expensive things).